### PR TITLE
Add new vulkan-memory-allocator package

### PR DESCRIPTION
**Pull request by felixaszx**
This new package provide a memory allcoator for Vulkan. And it is widely used as well.

### CODE REVIEW

**Comment by Biswa96** [comment]
You need to pin a specific commit of that repository. Please see the PKGBUILD-templates in this repository.

**Comment by felixxing** [comment]
> You need to pin a specific commit of that repository. Please see the PKGBUILD-templates in this repository.

ok, let me check

**Comment by felixxing** [comment]
> You need to pin a specific commit of that repository. Please see the PKGBUILD-templates in this repository.

I have specify a commit for the package.

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Maintainer: Felix Xing <felixaszx@outlook.com>
+
+_realname=vulkan-memory-allocator
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver="3.0.1"
+pkgrel=1
+pkgdesc="C++ user interface toolkit (mingw-w64)"
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:8]:
```suggestion
pkgdesc="Easy to integrate Vulkan memory allocation library (mingw-w64)"
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+arch=("any")
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
+url="https://www.fltk.org/"
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:11]:
```suggestion
url="https://gpuopen.com/vulkan-memory-allocator/"
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+license=('spdx:LGPL-2.0-or-later WITH FLTK-exception')
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:12]:
```suggestion
license=('spdx:MIT')
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+depends=("${MINGW_PACKAGE_PREFIX}-vulkan-loader")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-ninja"
+             "${MINGW_PACKAGE_PREFIX}-cc"
+             'git')
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:17]:
```suggestion
             "${MINGW_PACKAGE_PREFIX}-cc")
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+_commit='29d492b60c84ca784ea0943efc7d2e6e0f3bdaa'
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:18]:
```suggestion
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+source=("${_realname}"::"git+https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator.git#commit=${_commit}")
+sha256sums=('SKIP')
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:20]:
```suggestion
source=("${_realname}-${pkgver}"::"https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/archive/v${pkgver}.tar.gz")
sha256sums=('2a84762b2d10bf540b9dc1802a198aca8ad1f3d795a4ae144212c595696a360c')
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+prepare() {
+  cd "${srcdir}/${_realname}"
+}
```
Comment (MehdiChinoune) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:24]:
```suggestion
```

**Review by MehdiChinoune — CHANGES_REQUESTED** [review]

**Comment by felixxing** [comment]
Thank for change,  I just copy the old fltk build file.

**Comment by MehdiChinoune** [comment]
Would you please consider building shared library instead of static one by adding `-DBUILD_SHARED_LIBS=ON` to cmake flags.

**Comment by felixxing** [comment]
Sure. I will do that tomorrow. Its 12 in Eastern right now.

________________________________

**Comment by felixxing** [comment]
> Would you please consider building shared library instead of static one by adding `-DBUILD_SHARED_LIBS=ON` to cmake flags.

This is a header only library. I just get rid of the entire lib diretory. It doesn't do anything.

**Review by felixaszx — COMMENTED** [review]
Request review for finale changes of VMA.
This is now using the official VMA .zip release found on Github

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Maintainer: Felix Xing <felixaszx@outlook.com>
+
+_realname=VulkanMemoryAllocator
+pkgbase=mingw-w64-vulkan-memory-allocator
+pkgname="${MINGW_PACKAGE_PREFIX}-vulkan-memory-allocator"
+pkgver="3.0.1"
+pkgrel=1
+pkgdesc="Easy to integrate Vulkan memory allocation library (mingw-w64)"
```
Comment (filnet) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:8]: Should be `The industry-leading, open source, memory allocation library for the Vulkan API` ?

As per https://gpuopen.com/vulkan-memory-allocator/

Comment (felixaszx) [reply mingw-w64-vulkan-memory-allocator/PKGBUILD:8]: Its says `Easy to integrate Vulkan memory allocation library.` on github.
[https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator](url)
That's funny, I will change it to yours.

**Comment by felixaszx** [comment]
Request for review?

**Comment by rachit7645** [comment]
Any updates on this?

**Comment by felixaszx** [comment]
well, there should be.
Unfortunatelly I am a bit busy doing school work now.
I might try to fix any little gliches tonmorow

________________________________

**Review by felixaszx — COMMENTED** [review]
Its now ready for merge

**Pull request closed without merging**